rcl_action/src/rcl_action/types.c

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl_action/types.h"
+
+#include "rcl/error_handling.h"
+
+
+rcl_action_goal_info_t
+rcl_action_get_zero_initialized_goal_info(void)
+{
+  static rcl_action_goal_info_t goal_info = {
```

Comment:
@jacobperron I wonder if just `goal_info = {0}` would do for all compilers (as it does for gcc).

Reply:
Yes, that will work. For some reason I thought the internal uuid array wasn't initializing with the proper size.